internal/dataplane/parser/translate_httproute.go

```diff
@@ -235,7 +235,8 @@ func generateKongRoutesFromHTTPRouteMatches(
 		return []kongstate.Route{r}, nil
 	}
 
-	// TODO: implement query param matches (https://github.com/Kong/kubernetes-ingress-controller/issues/2778)
+	// Kong starts to support query parameter match since 3.4.1 with expression router
+	// so with traditional/traditional compatible router, we do not support query parameter match.
```

Comment:
Where does it say that this function is only for traditional or traditional compatible router? Shouldn't we check the parser feature flags? 🤔

Reply:
This function is for translating to traditional routes only. The check is done here: https://github.com/Kong/kubernetes-ingress-controller/blob/b3f5bf4482d3abd3874c5ef1a376d67e6753e9e0/internal/dataplane/parser/translate_httproute.go#L33. In later refactors of translators (possibly doing #4321), I will make this clearer.

Reply:
Got it. It's a bit hard to navigate through the code to figure this out.